libvast/test/module.cpp

```diff
@@ -0,0 +1,689 @@
+    return type{name, known_type};
+  }
+  const auto* decl_ptr = caf::get_if<record>(&declaration);
+  if (decl_ptr == nullptr)
+    return caf::make_error(ec::parse_error, "parses type alias with invalid "
+                                            "format");
+  const auto& decl = *decl_ptr;
+  // Get the optional attributes
+  auto attributes = std::vector<type::attribute_view>{};
+  auto found_attributes = decl.find("attributes");
+  if (found_attributes != decl.end()) {
+    const auto* attribute_list = caf::get_if<list>(&found_attributes->second);
+    if (attribute_list == nullptr)
+      return caf::make_error(ec::parse_error, "parses an attribute list with "
+                                              "an invalid format");
+    for (const auto& attribute : *attribute_list) {
+      const auto* attribute_string_ptr = caf::get_if<std::string>(&attribute);
+      if (attribute_string_ptr != nullptr)
+        attributes.push_back({*attribute_string_ptr});
+      else {
+        const auto* attribute_record_ptr = caf::get_if<record>(&attribute);
+        if (attribute_record_ptr == nullptr)
+          return caf::make_error(ec::parse_error, "parses an attribute with an "
+                                                  "invalid format");
+        const auto& attribute_record = *attribute_record_ptr;
+        if (attribute_record.size() != 1)
+          return caf::make_error(ec::parse_error, "parses an attribute not "
+                                                  "having one and only one "
+                                                  "field");
+        const auto& attribute_key = attribute_record.begin()->first;
+        const auto* attribute_value_ptr
+          = caf::get_if<std::string>(&attribute_record.begin()->second);
+        if (attribute_value_ptr == nullptr)
+          return caf::make_error(ec::parse_error, "parses an attribute with a "
+                                                  "non-string value");
+        const auto& attribute_value = *attribute_value_ptr;
+        attributes.push_back({attribute_key, attribute_value});
+      }
+    }
+  }
+  // Check that only one of type, enum, list, map and record is specified
+  // by the user
+  auto found_type = decl.find("type");
+  auto found_enum = decl.find("enum");
+  auto found_list = decl.find("list");
+  auto found_map = decl.find("map");
+  auto found_record = decl.find("record");
+  auto is_type_found = found_type != decl.end();
+  auto is_enum_found = found_enum != decl.end();
+  auto is_list_found = found_list != decl.end();
+  auto is_map_found = found_map != decl.end();
+  auto is_record_found = found_record != decl.end();
+  int type_selector_cnt = 0;
+  if (is_type_found)
+    type_selector_cnt++;
+  if (is_enum_found)
+    type_selector_cnt++;
+  if (is_list_found)
+    type_selector_cnt++;
+  if (is_map_found)
+    type_selector_cnt++;
+  if (is_record_found)
+    type_selector_cnt++;
```

Comment:
```suggestion
  int type_selector_cnt = is_type_found + is_enum_found + is_list_found + is_map_found + is_record_found;
```

Booleans convert to 0 or 1 in additions, which is a relict from C. Useful here, though, to check how many are true.